### PR TITLE
Add eslint plugin jsdoc [WIP]

### LINKED ISSUE #164: Integrate eslint plugin for jsdoc validation and fix potential issues
#### Reason/Context
We rely on JSDoc not only to generate API ref docs but now also to generate TypeScript types. We should take case for the quality of those.

#### Description
Please try answering few of those questions
- integrate https://www.npmjs.com/package/eslint-plugin-jsdoc with eslint
- fix potential issues

### PR DESCRIPTION
**Pull request by jonastg**
**Description**

Add JSDoc Eslint plugin with recommended configuration. Solved basic warnings, remaining warnings 295:

```
parser-js/lib/customValidators.js
  97:0  warning  Missing JSDoc @param "operations" description  jsdoc/require-param-description
  97:0  warning  Missing JSDoc @param "operations" type         jsdoc/require-param-type

parser-js/lib/errors/parser-error.js
  17:0  warning  Syntax error in namepath: module:@asyncapi/parser#ParserError                jsdoc/valid-types
  18:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  24:0  warning  Missing JSDoc @param "definition" description                                jsdoc/require-param-description
  58:3  warning  Missing JSDoc @returns declaration                                           jsdoc/require-returns

parser-js/lib/mixins/bindings.js
  10:0  warning  Missing JSDoc @returns description                      jsdoc/require-returns-description
  17:0  warning  Missing JSDoc @returns description                      jsdoc/require-returns-description
  24:0  warning  Missing JSDoc @returns description                      jsdoc/require-returns-description
  32:0  warning  Missing JSDoc @returns description                      jsdoc/require-returns-description
  40:0  warning  Invalid JSDoc @returns type "Object"; prefer: "object"  jsdoc/check-types
  40:0  warning  Missing JSDoc @returns description                      jsdoc/require-returns-description

parser-js/lib/mixins/description.js
  10:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  17:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description

parser-js/lib/mixins/external-docs.js
  12:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  19:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description

parser-js/lib/mixins/specification-extensions.js
   8:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  15:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  28:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  35:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  43:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  54:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  65:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  73:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description

parser-js/lib/mixins/tags.js
  10:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  17:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  24:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  32:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description
  40:0  warning  Missing JSDoc @returns description  jsdoc/require-returns-description

parser-js/lib/models/asyncapi.js
   22:0  warning  Syntax error in namepath: module:@asyncapi/parser#AsyncAPIDocument                                       jsdoc/valid-types
   42:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   49:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   56:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   63:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   70:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   78:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   85:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   92:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
   99:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  108:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  115:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  122:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  129:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  137:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  144:0  warning  The type 'Message' is undefined                                                                          jsdoc/no-undefined-types
  144:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  175:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  175:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  195:0  warning  Missing JSDoc @returns description                                                                       jsdoc/require-returns-description
  206:0  warning  Missing JSDoc @param "doc" description                                                                   jsdoc/require-param-description
  222:0  warning  Missing JSDoc @param "doc" description                                                                   jsdoc/require-param-description
  237:0  warning  Missing JSDoc @param "doc" description                                                                   jsdoc/require-param-description
  251:0  warning  Missing JSDoc @param "doc" description                                                                   jsdoc/require-param-description
  265:1  warning  Missing JSDoc @param "messages" declaration                                                              jsdoc/require-param
  265:1  warning  Missing JSDoc @param "number" declaration                                                                jsdoc/require-param
  269:0  warning  The type 'Message' is undefined                                                                          jsdoc/no-undefined-types
  269:0  warning  Expected @param names to be "messages, number". Got "map"                                                jsdoc/check-param-names
  279:1  warning  Missing JSDoc @returns declaration                                                                       jsdoc/require-returns
  280:0  warning  There must be a newline after the description of the JSDoc block                                         jsdoc/newline-after-description
  282:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  289:1  warning  Missing JSDoc @param "prop" declaration                                                                  jsdoc/require-param
  289:1  warning  Missing JSDoc @returns declaration                                                                       jsdoc/require-returns
  293:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  304:0  warning  There must be a newline after the description of the JSDoc block                                         jsdoc/newline-after-description
  307:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  308:0  warning  Expected JSDoc block to be aligned                                                                       jsdoc/check-alignment
  309:1  warning  Missing JSDoc @param "schemaContent" declaration                                                         jsdoc/require-param
  309:1  warning  Missing JSDoc @returns declaration                                                                       jsdoc/require-returns
  313:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  313:0  warning  Expected @param names to be "schemaContent, callback". Got "schema, callback"                            jsdoc/check-param-names
  314:0  warning  Missing JSDoc @param "callback" description                                                              jsdoc/require-param-description
  323:1  warning  Missing JSDoc @param "schema" declaration                                                                jsdoc/require-param
  323:1  warning  Missing JSDoc @param "callback" declaration                                                              jsdoc/require-param
  323:1  warning  Missing JSDoc @returns declaration                                                                       jsdoc/require-returns
  327:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  327:0  warning  Expected @param names to be "schema, seenObj, callback". Got "schemaContent, seenObj, callback(schema)"  jsdoc/check-param-names
  329:0  warning  Syntax error in namepath: callback(schema)                                                               jsdoc/valid-types
  368:0  warning  Missing JSDoc @param "doc" description                                                                   jsdoc/require-param-description
  369:0  warning  Missing JSDoc @param "callback" description                                                              jsdoc/require-param-description
  400:0  warning  Missing JSDoc @param "doc" description                                                                   jsdoc/require-param-description
  412:1  warning  Missing JSDoc @param "callback" declaration                                                              jsdoc/require-param
  416:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  418:0  warning  Syntax error in namepath: callback(schema)                                                               jsdoc/valid-types
  418:0  warning  Expected @param names to be "schema, seenObj, callback". Got "schema, seenObj, callback(schema)"         jsdoc/check-param-names
  435:1  warning  Missing JSDoc @param "callback" declaration                                                              jsdoc/require-param
  439:0  warning  The type 'Schema' is undefined                                                                           jsdoc/no-undefined-types
  441:0  warning  Syntax error in namepath: callback(schema)                                                               jsdoc/valid-types
  441:0  warning  Expected @param names to be "schema, seenObj, callback". Got "schema, seenObj, callback(schema)"         jsdoc/check-param-names

parser-js/lib/models/base.js
   7:0  warning  Syntax error in namepath: module:@asyncapi/parser#Base  jsdoc/valid-types
  16:3  warning  Missing JSDoc @param "key" declaration                  jsdoc/require-param
  17:0  warning  Missing JSDoc @returns description                      jsdoc/require-returns-description

parser-js/lib/models/channel-parameter.js
  13:0  warning  Syntax error in namepath: module:@asyncapi/parser#ChannelParameter           jsdoc/valid-types
  14:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  21:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  28:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/channel.js
  16:0  warning  Syntax error in namepath: module:@asyncapi/parser#Channel                    jsdoc/valid-types
  17:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  25:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  33:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  40:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  47:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  55:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  63:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  70:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/components.js
   18:0  warning  Syntax error in namepath: module:@asyncapi/parser#Components                 jsdoc/valid-types
   19:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
   25:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   31:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
   32:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   39:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   45:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
   46:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   53:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   59:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
   60:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   67:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   73:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
   74:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   81:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   87:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
   88:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   95:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  101:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
  102:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  109:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  115:3  warning  Missing JSDoc @param "name" declaration                                      jsdoc/require-param
  116:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/contact.js
  12:0  warning  Syntax error in namepath: module:@asyncapi/parser#Contact                    jsdoc/valid-types
  13:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  19:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  26:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  33:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/correlation-id.js
  12:0  warning  Syntax error in namepath: module:@asyncapi/parser#CorrelationId              jsdoc/valid-types
  13:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  20:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/external-docs.js
  12:0  warning  Syntax error in namepath: module:@asyncapi/parser#ExternalDocs               jsdoc/valid-types
  13:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  20:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/info.js
  14:0  warning  Syntax error in namepath: module:@asyncapi/parser#Info                       jsdoc/valid-types
  15:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  22:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  29:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  36:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  43:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  51:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/license.js
  11:0  warning  Syntax error in namepath: module:@asyncapi/parser#License                    jsdoc/valid-types
  12:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  18:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  25:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/message-trait.js
  7:0  warning  Syntax error in namepath: module:@asyncapi/parser#MessageTrait               jsdoc/valid-types
  8:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names

parser-js/lib/models/message-traitable.js
  17:0  warning  Syntax error in namepath: module:@asyncapi/parser#MessageTraitable           jsdoc/valid-types
  18:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  28:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  37:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  45:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  53:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  60:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  67:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  74:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  81:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  88:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/message.js
   8:0  warning  Syntax error in namepath: module:@asyncapi/parser#Message                    jsdoc/valid-types
   9:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  14:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  21:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  29:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  36:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/oauth-flow.js
  11:0  warning  Syntax error in namepath: module:@asyncapi/parser#OAuthFlow                  jsdoc/valid-types
  12:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  18:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  25:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  32:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  39:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/operation-trait.js
  7:0  warning  Syntax error in namepath: module:@asyncapi/parser#OperationTrait             jsdoc/valid-types
  8:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names

parser-js/lib/models/operation-traitable.js
  15:0  warning  Syntax error in namepath: module:@asyncapi/parser#OperationTraitable         jsdoc/valid-types
  16:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  26:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  33:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/operation.js
   8:0  warning  Syntax error in namepath: module:@asyncapi/parser#Operation                  jsdoc/valid-types
   9:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  14:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  23:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  31:3  warning  Missing JSDoc @param "index" declaration                                     jsdoc/require-param
  32:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/publish-operation.js
   7:0  warning  Syntax error in namepath: module:@asyncapi/parser#PublishOperation           jsdoc/valid-types
   8:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  13:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  20:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  27:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/schema.js
   13:0  warning  Syntax error in namepath: module:@asyncapi/parser#Schema                     jsdoc/valid-types
   14:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
   22:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   29:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   36:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   43:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   50:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   57:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   64:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   71:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   78:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   85:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   92:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
   99:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  106:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  113:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  120:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  127:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  134:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  141:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  148:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  156:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  164:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  172:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  180:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  191:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  198:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  208:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  217:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  224:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  231:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  239:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  251:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  259:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  267:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  275:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  283:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  290:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  297:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  304:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  311:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  318:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  325:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  332:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  338:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  345:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  352:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  359:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  366:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  373:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/security-scheme.js
  13:0  warning  Syntax error in namepath: module:@asyncapi/parser#SecurityScheme             jsdoc/valid-types
  14:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  21:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  28:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  35:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  42:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  49:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  56:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  63:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/server-security-requirement.js
  7:0  warning  Syntax error in namepath: module:@asyncapi/parser#ServerSecurityRequirement  jsdoc/valid-types
  8:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names

parser-js/lib/models/server-variable.js
  12:0  warning  Syntax error in namepath: module:@asyncapi/parser#ServerVariable             jsdoc/valid-types
  13:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  20:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  28:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  36:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  43:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  50:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  57:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/server.js
  15:0  warning  Syntax error in namepath: module:@asyncapi/parser#Server                     jsdoc/valid-types
  16:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  24:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  31:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  38:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  45:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  53:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  60:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  67:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/subscribe-operation.js
   4:0  warning  There must be a newline after the description of the JSDoc block             jsdoc/newline-after-description
   6:0  warning  Syntax error in namepath: module:@asyncapi/parser#SubscribeOperation         jsdoc/valid-types
   7:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names
  12:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  19:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description
  26:0  warning  Missing JSDoc @returns description                                           jsdoc/require-returns-description

parser-js/lib/models/tag.js
  13:0  warning  Syntax error in namepath: module:@asyncapi/parser#Tag                        jsdoc/valid-types
  14:0  warning  Invalid JSDoc tag (preference). Replace "extends" JSDoc tag with "augments"  jsdoc/check-tag-names

parser-js/lib/parser.js
   20:0  warning  Syntax error in namepath: @asyncapi/parser  jsdoc/valid-types
  139:1  warning  Missing JSDoc comment                       jsdoc/require-jsdoc
  161:1  warning  Missing JSDoc comment                       jsdoc/require-jsdoc
  167:1  warning  Missing JSDoc comment                       jsdoc/require-jsdoc
  179:1  warning  Missing JSDoc comment                       jsdoc/require-jsdoc
  214:1  warning  Missing JSDoc comment                       jsdoc/require-jsdoc

parser-js/lib/utils.js
  186:1  warning  Missing JSDoc @returns declaration      jsdoc/require-returns
  196:3  warning  Missing JSDoc comment                   jsdoc/require-jsdoc
  254:1  warning  Missing JSDoc @param "str" declaration  jsdoc/require-param
  254:1  warning  Missing JSDoc @returns declaration      jsdoc/require-returns
  266:1  warning  Missing JSDoc @param "arr" declaration  jsdoc/require-param
  266:1  warning  Missing JSDoc @param "obj" declaration  jsdoc/require-param
  266:1  warning  Missing JSDoc @returns declaration      jsdoc/require-returns

parser-js/test/mixins/bindings_test.js
  66:1  warning  Missing JSDoc comment  jsdoc/require-jsdoc

parser-js/test/mixins/description_test.js
  36:1  warning  Missing JSDoc comment  jsdoc/require-jsdoc

parser-js/test/mixins/external-docs_test.js
  40:1  warning  Missing JSDoc comment  jsdoc/require-jsdoc

parser-js/test/mixins/specification-extensions_test.js
  108:1  warning  Missing JSDoc comment  jsdoc/require-jsdoc

parser-js/test/mixins/tags_test.js
  78:1  warning  Missing JSDoc comment  jsdoc/require-jsdoc
```
**Related issue(s)**
Fixes #164

### CODE REVIEW

File: lib/customValidators.js [hunk 1]
```diff
@@ -90,10 +91,11 @@ function validateChannelParams(parsedJSON, asyncapiYAMLorJSON, initialFormat) {
...
  * 
  * @private
- * @param  {Object} parsedJSON parsed AsyncAPI document
- * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
- * @param  {String} initialFormat information of the document was oryginally JSON or YAML
- * @returns {Boolean} true in case the document is valid, otherwise throws ParserError
+ * @param {object} parsedJSON parsed AsyncAPI document
+ * @param {string} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param {string} initialFormat information of the document was oryginally JSON or YAML
+ * @param operations
```
Comment (derberg) [inline lib/customValidators.js:97]: this should be an array with names of all available channel operations, now it is `['publish', 'subscribe']`

**Comment by derberg** [comment]
@jonastg hey, do you plan to continue working on this one?

**Comment by jonastg** [comment]
@derberg yes, give me a couple of weeks to work on it

**Comment by derberg** [comment]
@jonastg sure, lemme know if ya needs some support, good luck.

**Comment by derberg** [comment]
looks like this one is discontinued, feel free to reopen when you are ready for the review

**Pull request closed without merging**